src/main/java/org/sonar/plugins/jacoco/KotlinFileLocator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SonarQube JaCoCo Plugin
+ * Copyright (C) 2018-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.jacoco;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class KotlinFileLocator {
+  private static final Logger LOGGER = Loggers.get(KotlinFileLocator.class);
+  private static final String SHEBANG_LINE = "#!.*+\n";
+  private static final String DELIMITED_COMMENT = "/\\*[^(*/)]*+\\*/";
+  private static final String LINE_COMMENT = "//[^\n]*+(\n|$)";
+  private static final String STRING_LITERAL = "\"[^\"]*+\"";
```

Comment:
I guess we don't consider escaped quotes in this regex. But I think that is not an issue for our use-case.